driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -1311,7 +1311,13 @@ protected void initChannel(SocketChannel channel) throws Exception {
             ChannelPipeline pipeline = channel.pipeline();
 
             if (sslOptions != null) {
-                pipeline.addLast("ssl", sslOptions.newSSLHandler(channel));
+                if (sslOptions instanceof AddrInfoSSLOptions) {
+                    InetSocketAddress addr = new InetSocketAddress(connection.address.getAddress(), connection.address.getPort());
```

Comment:
This seems unnecessary, `connection.address` is already an `InetSocketAddress`, so the following should suffice:

```java
pipeline.addLast("ssl", sslOptions.newSSLHandler(channel, connection.address));
```